charts/seed-monitoring/charts/grafana/dashboards/operators/vpn-seed-server/vpn-seed-server-dashboard.json

```diff
@@ -0,0 +1,1188 @@
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "thresholds"
+          },
+          "custom": {
+            "align": null,
+            "displayMode": "auto",
+            "filterable": false
+          },
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              }
+            ]
+          }
+        },
+        "overrides": [
+          {
+            "matcher": {
+              "id": "byName",
+              "options": "Connection Time"
+            },
+            "properties": [
+              {
+                "id": "unit",
+                "value": "dateTimeAsIso"
+              }
+            ]
+          },
+          {
+            "matcher": {
+              "id": "byName",
+              "options": "Rx"
+            },
+            "properties": [
+              {
+                "id": "unit",
+                "value": "bytes"
+              }
+            ]
+          },
+          {
+            "matcher": {
+              "id": "byName",
+              "options": "Tx"
+            },
+            "properties": [
+              {
+                "id": "unit",
+                "value": "bytes"
+              }
+            ]
+          },
+          {
+            "matcher": {
+              "id": "byName",
+              "options": "Last seen"
+            },
+            "properties": [
+              {
+                "id": "unit",
+                "value": "dateTimeAsIso"
+              }
+            ]
+          }
+        ]
+      },
+      "gridPos": {
+        "h": 13,
+        "w": 14,
+        "x": 10,
+        "y": 30
+      },
+      "id": 4,
+      "links": [],
+      "options": {
+        "showHeader": true,
+        "sortBy": [
+          {
+            "desc": false,
+            "displayName": "Server"
+          }
+        ]
+      },
+      "pluginVersion": "7.5.17",
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "sum(label_replace(label_replace(openvpn_server_client_sent_bytes_total{},\"real_ip\", \"$1\", \"real_address\", \"(.+):.+\"), \"service\", \"$1\", \"instance\", \"([^.]+).+\")) by(common_name, connection_time, virtual_address,real_ip,real_address,service)",
+          "format": "table",
+          "instant": true,
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "",
+          "refId": "B"
+        },
+        {
+          "exemplar": true,
+          "expr": "sum(label_replace(label_replace(openvpn_server_client_received_bytes_total{},\"real_ip\", \"$1\", \"real_address\", \"(.+):.+\"), \"service\", \"$1\", \"instance\", \"([^.]+).+\")) by(common_name, connection_time, virtual_address,real_ip,real_address,service)",
```

Comment:
Assuming that the real_address label contains the TCP port, this could significantly increase the cardinality of the newly scraped metrics. Can we estimate the load on Prometheus in a regular shoot cluster? A mitigation / preemptive measure could be to apply relabel configs to remove the labels with high cardinality. Maybe we could check this together in a dev environment.

Reply:
These `openvpn_server_*` metrics are collected from the vpn-seed-server pods only (at the moment two per control plane). Additionally there are only a few clients (typically 5 - 8: two clients on the shoot side plus the number of kube-apiserver pods on the seed side). It is not depending on the number of nodes in the shoot cluster.